### ISSUE TITLE
Source Control View has no backoff delay on authorization errors

### ISSUE THREAD

**Issue by hyldmo** [microsoft/vscode#83098]
While opening a folder/repo where vscode is not able to authenticate, it seems it will instantly retry indefinitely. In this case, this lead to me being IP-banned from Gitlab, leaving many of my repositories inaccessible (hopefully only for a short while).

There should probably be something like an exponential backoff delay, or that it eventually gives up so that it doesn't spam servers.

Here's an excerpt of log from the Git tab in the Output window to illustrate:
```
Open repository: d:\Mega\dev\auk\auk-utils
> git fetch
> git status -z -u
> git rev-parse --show-toplevel
> git rev-parse --git-dir
> git config --get commit.template
Open repository: d:\Mega\dev\auk\auk-utils
> git fetch
> git status -z -u
remote: Forbidden
fatal: unable to access 'https://gitlab.com/matauk/auk-utils.git/': The requested URL returned error: 403
> git rev-parse --show-toplevel
> git rev-parse --git-dir
> git config --get commit.template
Open repository: d:\Mega\dev\auk\auk-utils
> git fetch
> git status -z -u
remote: Forbidden
fatal: unable to access 'https://gitlab.com/matauk/auk-utils.git/': The requested URL returned error: 403
remote: Forbidden
fatal: unable to access 'https://gitlab.com/matauk/auk-utils.git/': The requested URL returned error: 403
remote: Forbidden
fatal: unable to access 'https://gitlab.com/matauk/auk-utils.git/': The requested URL returned error: 403
> git rev-parse --show-toplevel
> git rev-parse --git-dir
> git config --get commit.template
Open repository: d:\Mega\dev\auk\auk-utils
> git fetch
> git status -z -u
remote: Forbidden
fatal: unable to access 'https://gitlab.com/matauk/auk-utils.git/': The requested URL returned error: 403
> git rev-parse --show-toplevel
> git rev-parse --git-dir
> git config --get commit.template
Open repository: d:\Mega\dev\auk\auk-utils
...
```

**Comment by hyldmo** [comment]
Yeah, it looks like it should stop trying when it hits the auth error, or wait 3 minutes, but here it retries instantly. 
I think I discovered why it seems to reopen constantly though. I saw I had a broken submodule (the kind of bug that happens when you switches to a branch without the submodule, then switch back again) inside the repo, and when I fixed that submodule it stopped with the instant retrying. 

No idea what could be causing it though, the code you linked looks pretty bulletproof. 

Could it be that the broken submodule causes https://github.com/microsoft/vscode/blob/a143fc36f8ee58db069c0bd2a14ec4ddeb29a009/extensions/git/src/autofetch.ts#L113 to fail silently, returning 0 instead of throwing? That would make the code retry instantly. Still weird that the auth error is not caught though.
